test/e2erp/realrp_test.go

```diff
@@ -14,4 +23,72 @@ var _ = Describe("Resource provider e2e tests [Real]", func() {
 		var err error
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("should not be possible for customer to mutate an osa scale set", func() {
+
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+		logger := logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": c.resourceGroup})
+
+		arg := azureclient.ApplicationResourceGroup(c.resourceGroup, c.resourceGroup, c.location)
+		Expect(arg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("application resource group is %s", arg)
+
+		// update the logger to use the application resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": arg})
+
+		mrg, err := azureclient.ManagedResourceGroup(ctx, arg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(arg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("managed resource group is %s", mrg)
+
+		// update the logger to use the managed resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": mrg})
+
+		scaleSets, err := ScaleSets(ctx, logger, c.ssc, mrg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(scaleSets).NotTo(And(BeNil(), BeEmpty()))
+		Expect(len(scaleSets)).Should(BeNumerically(">=", 3))
```

Comment:
Are customers able to see all of these resources? If so, then this should be == 3, right?

Reply:
Yes it could be set equal to 3 but `>= 3` allows this to work when we have more than a group of compute nodes (multiple compute node profiles) in the future. The idea is that each clusters needs at least 1 master scale set, 1 infra scale set and 1 compute scale set.

Reply:
Sorry...I don't agree with this. There's nothing wrong with having a look ahead and planning or writing code to take future situations into account as long as the present is also taken into account.